src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -130,6 +130,15 @@ Copyright (c) .NET Foundation. All rights reserved.
       />
   </Target>
 
+  <!--
+    Trigger an error if C++ project targeting a lower version of .NET Core than 3.1.
+  -->
+  <Target Name="_CheckForUnsupportedCppNETCoreVersion" BeforeTargets="_CheckForInvalidConfigurationAndPlatform;Restore;CollectPackageReferences">
```

Comment:
Why are these the only 3 'BeforeTargets':

1. _CheckForInvalidConfigurationAndPlatform
2. Restore
3. CollectPackageReferences

I would think there would be many more...

Reply:
I copied it from a similar task above https://github.com/dotnet/sdk/blob/63fa4e00148887ff50885f6068a677893def6fd0/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets#L124

And for C++, pretty much only build and restore is supported. These covered them

Reply:
This looks ok to me. `_CheckForInvalidConfigurationAndPlatform` is run for ~all build scenarios.